openstack/tools/openstack_regression_tests_runner.sh

```diff
@@ -0,0 +1,143 @@
+else
+    git clone https://github.com/openstack-charmers/zosci-config
+fi
+)
+
+TOOLS_PATH=$(realpath $(dirname $0))/func_test_tools
+CHARM_PATH=$(pwd)
+
+echo "Running regression tests"
+
+source ~/novarc
+export {,TEST_}CIDR_EXT=`openstack subnet show subnet_${OS_USERNAME}-psd-extra -c cidr -f value`
+FIP_MAX=$(ipcalc $CIDR_EXT| awk '$1=="HostMax:" {print $2}')
+FIP_MIN=$(ipcalc $CIDR_EXT| awk '$1=="HostMin:" {print $2}')
+FIP_MIN_ABC=${FIP_MIN%.*}
+FIP_MIN_D=${FIP_MIN##*.}
+FIP_MIN=${FIP_MIN_ABC}.$(($FIP_MIN_D + 64))
+
+CIDR_OAM=`openstack subnet show subnet_${OS_USERNAME}-psd -c cidr -f value`
+OAM_MAX=$(ipcalc $CIDR_OAM| awk '$1=="HostMax:" {print $2}')
+OAM_MIN=$(ipcalc $CIDR_OAM| awk '$1=="HostMin:" {print $2}')
+OAM_MIN_ABC=${OAM_MIN%.*}
+OAM_MAX_D=${OAM_MAX##*.}
+# Picking last two addresses and hoping they dont get used by Neutron.
+export {OS,TEST}_VIP00=${OAM_MIN_ABC}.$(($OAM_MAX_D - 1))
+export {OS,TEST}_VIP01=${OAM_MIN_ABC}.$(($OAM_MAX_D - 2))
+
+# More information on config https://github.com/openstack-charmers/zaza/blob/master/doc/source/runningcharmtests.rst
+export {,TEST_}NET_ID=$(openstack network show net_${OS_USERNAME}-psd-extra -f value -c id)
+export {,TEST_}FIP_RANGE=$FIP_MIN:$FIP_MAX
+export {,TEST_}GATEWAY=$(openstack subnet show subnet_${OS_USERNAME}-psd-extra -c gateway_ip -f value)
+export {,TEST_}NAME_SERVER=91.189.91.131
+export {,TEST_}CIDR_PRIV=192.168.21.0/24
+export {,TEST_}SWIFT_IP=10.140.56.22
+export TEST_MODEL_SETTINGS="image-stream=released;default-series=jammy;test-mode=true;transmit-vendor-metrics=false"
+# We need to set TEST_JUJU3 as well as the constraints file
+# Ref: https://github.com/openstack-charmers/zaza/blob/e96ab098f00951079fccb34bc38d4ae6ebb38606/setup.py#L47
+export TEST_JUJU3=1
+
+# NOTE: this should not be necessary for > juju 2.x but since we still have a need for it we add it in
+export TEST_ZAZA_BUG_LP1987332=1
+
+# Some charms point to an upstream constraints file that installs python-libjuju 2.x so we need to do this to ensure we get 3.x
+export TEST_CONSTRAINTS_FILE=https://raw.githubusercontent.com/openstack-charmers/zaza/master/constraints-juju34.txt
+
+LOGFILE=$(mktemp --suffix=-charm-func-test-results)
+(
+    # Ensure charmed-openstack-tester checked out and up-to-date
+    if [[ -d $HOME/charmed-openstack-tester ]]; then
+        cd $HOME/charmed-openstack-tester
+        git checkout master
+        git pull
+    else
+        cd $HOME
+        git clone https://github.com/openstack-charmers/charmed-openstack-tester
+        cd charmed-openstack-tester
+    fi
+
+    # If a func test pr is provided switch to that pr.
+    if [[ -n $FUNC_TEST_PR ]]; then
+        # We use the zosci-config tools to do this.
+        MSG=$(echo "Func-Test-Pr: https://github.com/openstack-charmers/zaza-openstack-tests/pull/$FUNC_TEST_PR"| base64)
+        ~/zosci-config/roles/handle-func-test-pr/files/process_func_test_pr.py -f ./test-requirements.txt "$MSG"
+    fi
+
+    fail=false
```

Comment:
What's this `fail` for?

Reply:
hangover from copy+paste have cleaned up